ubuntu/debian/libgz-transport.install

```diff
@@ -1 +1,4 @@
 usr/lib/*/lib*-transport[0-9][0-9].so.*
+usr/lib/ruby/*/cmdtransport[0-9][0-9].rb
+usr/share/*/transport[0-9][0-9].yaml
```

Comment:
I tested locally by adding the `usr/libexec` path like so:
```suggestion
usr/share/*/transport[0-9][0-9].yaml
usr/libexec/gz/transport[0-9][0-9]/*
```
and it seems to work. In the https://github.com/gazebo-release/gz-transport11-release/pull/4, we create a new package called `ignition-transport11-cli`, add it as a dependency of `libignition-transport11-dev`. However, we now want these binaries to be part of the non-dev package, but we can't add a dependency of `gz-transport12-cli` onto `libgz-transport12` since that will create a dependency cycle. I propose we just add these binaries to `libgz-transport12` and drop the `cli` package. @j-rivero thoughts?

Reply:
I believe we need the CLI to be in a non-lib package, because that's now installing a binary, even though it is a `libexec` one, and not `bin`

Reply:
> In the [gazebo-release/gz-transport11-release#4](https://github.com/gazebo-release/gz-transport11-release/pull/4), we create a new package called `ignition-transport11-cli`, add it as a dependency of `libignition-transport11-dev`. However, we now want these binaries to be part of the non-dev package, 

Let's stop here. Let me summarize the situation of the software pieces: we have a software composed by: main lib (libgz-transport12-dev), developer headers + helpers (libgz-transport12-dev and a cli program (gz-transport12-cli). The (hard -> `Depends:` in Debian) dependencies should be strict in the way that the package is not going to work without the dep. So the cli program probably depends on the lib but the lib should not depend on cli package since it does not need it. We can use other sotf dependencies ([Suggest, Recommends](https://www.debian.org/doc/debian-policy/ch-relationships.html)) to handle the behavior we want. So my proposal would be:
 * Keep the cli package (useful), make it `Depends` on the main lib
 * If we want people to install the CLI together with the lib by default, make the cli a `Suggest` of the main lib. 
 * The development package should depend on the lib + cli or we can move the cli to be also a `Suggest` (final behavior is the same, package manager install it together, latter seems to me like a more correct approach).

> I believe we need the CLI to be in a non-lib package, because that's now installing a binary, even though it is a `libexec` one, and not `bin`

If we are strict with the [Debian policy about shared libraries](https://www.debian.org/doc/debian-policy/ch-sharedlibs.html), then yes, they should into their own individual package so you can have two library packages from different versions installed together.

Reply:
Sounds good. @scpeters I can make the changes, if that's okay.

Reply:
please do so, thanks for taking this over